xbmc/addons/interfaces/GUI/Window.cpp

```diff
@@ -0,0 +1,665 @@
+}
+
+void CGUIAddonWindow::Update()
+{
+}
+
+void CGUIAddonWindow::AddItem(CFileItemPtr fileItem, int itemPosition)
+{
+  if (itemPosition == -1 || itemPosition > m_vecItems->Size())
+  {
+    m_vecItems->Add(fileItem);
+  }
+  else if (itemPosition <  -1 &&  !(itemPosition-1 < m_vecItems->Size()))
+  {
+    m_vecItems->AddFront(fileItem,0);
+  }
+  else
+  {
+    m_vecItems->AddFront(fileItem, itemPosition);
+  }
+  m_viewControl.SetItems(*m_vecItems);
+  UpdateButtons();
+}
+
+void CGUIAddonWindow::RemoveItem(int itemPosition)
+{
+  m_vecItems->Remove(itemPosition);
+  m_viewControl.SetItems(*m_vecItems);
+  UpdateButtons();
+}
+
+void CGUIAddonWindow::RemoveItem(CFileItem* fileItem)
+{
+  m_vecItems->Remove(fileItem);
+  m_viewControl.SetItems(*m_vecItems);
+  UpdateButtons();
+}
+
+int CGUIAddonWindow::GetCurrentListPosition()
+{
+  return m_viewControl.GetSelectedItem();
+}
+
+void CGUIAddonWindow::SetCurrentListPosition(int item)
+{
+  m_viewControl.SetSelectedItem(item);
+}
+
+int CGUIAddonWindow::GetListSize()
+{
+  return m_vecItems->Size();
+}
+
+CFileItemPtr CGUIAddonWindow::GetListItem(int position)
+{
+  if (position < 0 || position >= m_vecItems->Size()) return CFileItemPtr();
+  return m_vecItems->Get(position);
+}
+
+void CGUIAddonWindow::ClearList()
+{
+  ClearFileItems();
+
+  m_viewControl.SetItems(*m_vecItems);
+  UpdateButtons();
+}
+
+void CGUIAddonWindow::GetContextButtons(int itemNumber, CContextButtons& buttons)
+{
+  gui_context_menu_pair c_buttons[ADDON_MAX_CONTEXT_ENTRIES];
+  unsigned int size = ADDON_MAX_CONTEXT_ENTRIES;
+  if (CBGetContextButtons)
+  {
+    CBGetContextButtons(m_clientHandle, itemNumber, c_buttons, &size);
+    for (unsigned int i = 0; i < size; ++i)
+    {
+      buttons.push_back(std::pair<unsigned int, std::string>(c_buttons[i].id, c_buttons[i].name));
+    }
+  }
+}
+
+bool CGUIAddonWindow::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
+{
+  if (CBOnContextButton)
+    return CBOnContextButton(m_clientHandle, itemNumber, static_cast<unsigned int>(button));
+  return false;
+}
+
+void CGUIAddonWindow::WaitForActionEvent(unsigned int timeout)
+{
+  m_actionEvent.WaitMSec(timeout);
+  m_actionEvent.Reset();
+}
+
+void CGUIAddonWindow::PulseActionEvent()
+{
+  m_actionEvent.Set();
+}
+
+bool CGUIAddonWindow::OnClick(int item, const std::string& player)
+{
+  // Hook Over calling  CGUIMediaWindow::OnClick(iItem) results in it trying to PLAY the file item
+  // which if its not media is BAD and 99 out of 100 times undesireable.
+  return false;
+}
+
+void CGUIAddonWindow::SetupShares()
+{
+  UpdateButtons();
+}
+
+
+CGUIAddonWindowDialog::CGUIAddonWindowDialog(int id, const std::string& strXML, CAddonDll* addon)
+: CGUIAddonWindow(id, strXML, addon, false)
+{
+  m_bRunning = false;
+  m_bIsDialog = true;
+}
+
+bool CGUIAddonWindowDialog::OnMessage(CGUIMessage& message)
+{
+  if (message.GetMessage() == GUI_MSG_WINDOW_DEINIT)
+    return CGUIWindow::OnMessage(message);
```

Comment:
Has looked, goes back to the annals since them come in.

Maybe it was related to the old `ShowOverlay` call:
```cpp
bool CGUIAddonWindowDialog::OnMessage(CGUIMessage &message)
{
  if (message.GetMessage() == GUI_MSG_WINDOW_DEINIT)
  {
    CGUIWindow *pWindow = g_windowManager.GetWindow(g_windowManager.GetActiveWindow());
    if (pWindow)
      g_windowManager.ShowOverlay(pWindow->GetOverlayState());
    return CGUIWindow::OnMessage(message);
  }
  return CGUIAddonWindow::OnMessage(message);
}
```

I check and remove it complete, if not needed.